### PR TITLE
update demo to use ECDSA signing algorithm instead of RSA

### PR DESCRIPTION
**Pull request by andrewhalle**
* Update `setup.py` to generate an EC key
* Update `app.py` to use ECDSA signing algorithm

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,14 +18,14 @@
...
+from cryptography.hazmat.primitives.asymmetric import ec
 from pathlib import Path
 import json
 
 
 start_marker = '-----BEGIN CERTIFICATE REQUEST-----'
 end_marker = '-----END CERTIFICATE REQUEST-----'
-hash_alg = 'RSASSA_PKCS1_V1_5_SHA_256'
+hash_alg = 'ECDSA_SHA_256'
 sign_oid = '1.2.840.113549.1.1.11'
```
Comment (mauricefisher64) [inline setup.py:29]: I think this should probably be 1.2.840 .10045 .2 .1 (EC_PUBLICKEY_OID)

File: setup.py [hunk 2]
```diff
@@ -67,9 +67,8 @@ def generate_certificate_request(kms_key: str, subject: str):
     pubkey_der = response['PublicKey']
 
     # Create temporary private key
-    private_key = rsa.generate_private_key(
-        public_exponent=65537,
-        key_size=2048,
+    private_key = ec.generate_private_key(
+        ec.SECP384R1()
```
Comment (mauricefisher64) [inline setup.py:71]: The should be whatever maps to PRIME256V1

Comment (mauricefisher64) [reply setup.py:71]: Since you are using NIST the name could be secp256r1

File: setup.py (hunk 1 above)
Comment (andrewhalle) [reply setup.py:29]: Later on in this file, I use `1.2.840.10045.4.3.2`, what's the difference between that and the ID you've specified, and which should be preferred?

File: setup.py (hunk 2 above)
Comment (andrewhalle) [reply setup.py]: Updated to use `ec.SECP256R1()`

**Review by mauricefisher64 — APPROVED** [review]
👍

File: setup.py (hunk 1 above)
Comment (mauricefisher64) [reply setup.py]: 1.2.840.10045.4.3.2 is the signature type (ECDSA_WITH_SHA256_OID).  The other is the type of key (EC_PUBLICKEY_OID)

**Pull request merged**